packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -104,10 +124,90 @@ export enum TreatMissingData {
   MISSING = 'missing'
 }
 
+/**
+ * Enumeration indicates state of Alarm used in building Alarm Rule.
+ */
+export enum AlarmState {
+
+  /**
+   * State indicates resource is in ALARM
+   */
+  ALARM = 'ALARM',
+
+  /**
+   * State indicates resource is not in ALARM
+   */
+  OK = 'OK',
+
+  /**
+   * State indicates there is not enough data to determine is resource is in ALARM
+   */
+  INSUFFICIENT_DATA = 'INSUFFICIENT_DATA',
+
+}
+
+/**
+ * The base class for Alarm and CompositeAlarm resources.
+ */
+export abstract class AlarmBase extends Resource implements IAlarm {
```

Comment:
I think better not `export` this.

Reply:
i couldn't find a means of using this in ```composite-alarm.ts``` without exporting. my TS knowledge is very limited, one option i can think of is moving the ```CompositeAlarm``` implementation into alarm.ts. please advise.

Reply:
Hmm yeah. It needs to be in a separate file then (`alarm-base.ts`), and that new file needs to be NOT exported from `index.ts`.

Reply:
moved it to a separate file but had to export it since JSII complains unless exported

```
lib/alarm-action.ts:14:26 - error TS9999: JSII: Unable to resolve referenced type '@aws-cdk/aws-cloudwatch.IAlarm'. Type may be @internal or unexported
```